src/optimizer/rules/implementation_rules.cpp

```diff
@@ -487,11 +525,16 @@ void LogicalInnerJoinToPhysicalInnerIndexJoin::Transform(
   std::vector<AnnotatedExpression> join_preds = r_child->GetPredicates();
   for (auto &pred : inner_join->GetJoinPredicates()) join_preds.push_back(pred);
 
+  // Get inner join limit information
+  auto limit_exists = inner_join->GetLimitExists();
+  auto limit = inner_join->GetLimit();
+
   std::vector<std::unique_ptr<AbstractOptimizerNode>> empty;
+  auto child_get = LogicalGet::Make(r_child->GetDatabaseOid(), r_child->GetTableOid(), join_preds,
+                                    r_child->GetTableAlias(), r_child->GetIsForUpdate());
+  if (limit_exists) child_get.GetContentsAs<LogicalGet>()->SetLimit(limit);
```

Comment:
Just thinking out loud here, is it ever possible that we'll need more rows than the limit from the `child_get` to have enough rows in the join to satisfy the limit? I don't think so because we're also pushing the join predicates down and I haven't actually found any query that brakes because of this.

Reply:
I think it should be ok since this get is effectively a probe for after the join predicates, and the `LogicalGet` implementation rule should ignore the `Limit` in cases where all predicates aren't tracked by the index.